### DISCUSSION TITLE
Version: 3.0.3,revision:37c2389

### DISCUSSION THREAD

**Discussion by Terracotta-6** [artisan-roaster-scope/artisan#1721] [General]
File "artisanlib\main.py", line 4195, in pidOn File "artisanlib\pid_control.py", line 1395, in pidOn File .

[artisan.log.txt](https://github.com/user-attachments/files/17467954/artisan.log.txt)

**Comment by MAKOMO** [comment]
Thanks! Will be fixed on next commit.

**Reply by MAKOMO** [reply]
The commit is in and the issue should be fixed. THANKS FOR REPORTING!

**Reply by Terracotta-6** [reply]
Thank you for your so efficient work!

**Comment by Terracotta-6** [comment]
Artisan v3.0.3 (9b1ba5e, 0)    ERROR  : Error:   File "artisanlib\main.py", line 4195, in pidOn 

![4494902bbb7c5c7b3e63e53c870501c](https://github.com/user-attachments/assets/0f38a16c-2ce7-4a1f-b0bb-dcdaaced02a6)

[artisan.log](https://github.com/user-attachments/files/17495129/artisan.log)

**Reply by MAKOMO** [reply]
Wow! Another potential index error I overlooked in the very same function. I just fixed this one too. A new continuous build is on its way. Might take another 20min until it is completely available for download. Thanks for this additional testing!

**Comment by Terracotta-6** [comment]
message: setNsv(): bad response


https://github.com/user-attachments/assets/1492d89f-55a4-449d-8d04-91c43e2fdc9c

**Comment by MAKOMO** [comment]
Thanks for your report. This is very outdated code written more than 10 years by the original maintainer of Artisan. I checked the code and the documentation of the PXF. I also verified with my simulator that Artisan is sending the correct request. However, your device seems to deliver an unexpected answer. I cannot do anything about that as I do not have access to such a device to test further. Sorry.

**Comment by Terracotta-6** [comment]
designer load point error.
[artisan.log](https://github.com/user-attachments/files/17686777/artisan.log)

**Comment by MAKOMO** [comment]
Could you also attach the file that you tried to load the points from?

**Comment by Terracotta-6** [comment]
As a side note, an error occurs in saving the loading times of *.adsg files after dragging and modifying any control points during editing.
[2024-11.zip](https://github.com/user-attachments/files/17686886/2024-11.zip)

https://github.com/user-attachments/assets/04c18aa0-a85a-4232-8ad4-cc53e7a3b261
